### PR TITLE
feat: create rest service for user operations

### PR DESCRIPTION
**Pull request by MelihSelamiUrkmezz**
A rest service has been created for user operations with H2 db.

### CODE REVIEW

File: src/main/java/com/melihsurkmez/secondhand/ecommerce/DTO/CreateUserRequest.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.melihsurkmez.secondhand.ecommerce.DTO;
+
+import lombok.Data;
+
+@Data
```
Comment (fehmisener) [inline src/main/java/com/melihsurkmez/secondhand/ecommerce/DTO/CreateUserRequest.java:5]: Nice to use it 👍

File: src/main/java/com/melihsurkmez/secondhand/ecommerce/model/User.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import jakarta.persistence.*;
+import lombok.*;
+
+@Data
+@NoArgsConstructor
+@RequiredArgsConstructor
+@AllArgsConstructor
+@Entity
+@Table(name="users")
+public class User {
```
Comment (fehmisener) [inline src/main/java/com/melihsurkmez/secondhand/ecommerce/model/User.java:12]: Did you forget to use `BaseEntity`?

File: src/main/java/com/melihsurkmez/secondhand/ecommerce/service/UserService.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        User user= new User(newUser.getFirstName(),newUser.getMiddleName(),newUser.getLastName(),newUser.getEmail());
+
+        return userResponseConverter.convert(userRepository.save(user));
+
+    }
+
+    public UserResponse updateUser(String email,UpdateUserRequest updatedUser) {
+
+        User user = findUserByEmail(email);
+        User updatedNewUser=new User(user.getId(),updatedUser.getFirstName(),updatedUser.getMiddleName(),updatedUser.getLastName(),email);
```
Comment (fehmisener) [inline src/main/java/com/melihsurkmez/secondhand/ecommerce/service/UserService.java:50]: Why do you need to create a new `User updatedNewUser` object? Can't you use the object that you find with `findUserByEmail`? In fact, write a mapper for this purpose but be careful and do not forget to check null or empty variables.

```.jaav
User user = findUserByEmail(email);
user.setFirstName(updatedUser.getFirstName());
user.setMiddleName(updatedUser.getMiddleName());
...
```

**Review by fehmisener — APPROVED** [review]

File: src/main/java/com/melihsurkmez/secondhand/ecommerce/service/UserService.java (hunk 3 above)
Comment (MelihSelamiUrkmezz) [reply src/main/java/com/melihsurkmez/secondhand/ecommerce/service/UserService.java:50]: image

I assigned the updated user to the user and saved it again.

File: src/main/java/com/melihsurkmez/secondhand/ecommerce/model/User.java (hunk 2 above)
Comment (MelihSelamiUrkmezz) [reply src/main/java/com/melihsurkmez/secondhand/ecommerce/model/User.java:12]: I didnt forget. This class has not been used in 'FolksDev-Spring Boot Ecommerce' yet.

**Pull request merged**